Automattic/sensei
# Add Sensei Pro promo backend for Sensei Home.

Fixes #5744 

### Changes proposed in this Pull Request
* Changed interface for promotional banner so that it is a little bit more generic and avoid any references to sensei pro.
* Logic to show/hide banner is based on a new filter `sensei_home_promo_banner_show`.

### Testing instructions
- Run curl similar to: `curl -sX GET --user aaron:development --insecure 'https://test2.local/wp-json/sensei-internal/v1/home' | jq .promo_banner`. ([Check here to make curl authentication easier in local](https://evergreenp2.wordpress.com/2022/03/03/testing-cheatsheet/#test-rest-api-in-local-easily))
- Check that output is as expected (visible by default).
- Disable `sensei_home_promo_banner_show` filter with a snippet.
- Run curl again.
- Check that output is correct (not visible).

### Screenshot / Video
**By default**
```
$ curl -sX GET --user aaron:development --insecure 'https://test2.local/wp-json/sensei-internal/v1/home' | jq .promo_banner
{
  "is_visible": true
}
```
**With filter disabled**
```
$ curl -sX GET --user aaron:development --insecure 'https://test2.local/wp-json/sensei-internal/v1/home' | jq .promo_banner
{
  "is_visible": false
}
```
